salt/modules/rpm.py

```diff
@@ -338,10 +357,13 @@ def file_list(*packages):
         salt '*' lowpkg.file_list httpd postfix
         salt '*' lowpkg.file_list
     '''
+    cmd = ['rpm']
+    if 'root' in kwargs:
+        cmd.extend(['--root', kwargs['root']])
     if not packages:
-        cmd = ['rpm', '-qla']
+        cmd.append('-qla')
     else:
-        cmd = ['rpm', '-ql']
+        cmd.append('-ql')
```

Comment:
A bit spaggettish, maybe we can always add `-ql` and only append another `-a` when no packages? Or so:

```python
cmd.append('-ql{}'.format(packages and 'a' or ''))
```

Or like you did later:

```python
cmd.append('-ql' if packages else '-qla')
```